frontend/src/metabase-lib/parameters/utils/parameter-values.js

```diff
@@ -9,12 +9,25 @@ import {
 export const PULSE_PARAM_EMPTY = null;
 export const PULSE_PARAM_USE_DEFAULT = undefined;
 
-export function getValuePopulatedParameters(parameters, parameterValues) {
-  return parameters.map(parameter => ({
-    ...parameter,
-    value: parameterValues?.[parameter.id] ?? null,
+// defaultRequired must be used in features that support required parameters
+export function getParameterValue(parameter, values, defaultRequired = false) {
+  return (
+    values?.[parameter.id] ??
+    (defaultRequired && parameter.required ? parameter.default : null)
```

Comment:
I think having a few if blocks would read better than trying to cramp everything into a single expression. This took me some time to understand it.